### PR TITLE
Ruby - Testing your Ruby code - Tic Tac Toe

### PR DESCRIPTION
**Pull request by alvarorf**
The purpose of this project is to use Rspec to implement tests for the classes Player and Board in the game_logic.rb file and, since all of the methods belonging to such classes are public, we tested all of them.

### CODE REVIEW

File: spec/game_logic_spec.rb [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    end
+
+    it 'After updating the board, the cell should not be empty' do
+      expect do
+        let(:new_board) { Board.new }
+        new_board.update('5', 'O')
+        expect(new_board.rows[1][1]).to_not eq('_')
+      end
+    end
+  end
```
Comment (voscarmv) [inline spec/game_logic_spec.rb:60]: I noticed here you're not really using your `#valid_cell` method anywhere. Please call your `#valid_cell` method from this test in other to actually test it.

File: spec/game_logic_spec.rb [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+
+    it 'Should return false if the argument is not a valid number between 1 and 9' do
+      expect do
+        let(:new_board) { Board.new }
+        new_board.sanitize_choice('11')
+        expect(new_board.sanitize_choice).be_false
+      end
+    end
+  end
+end
```
Comment (voscarmv) [inline spec/game_logic_spec.rb:199]: Please also test `Player` initialization by testing if the player data is saved once the player is created. Also, add negative cases for such tests.

**Review by voscarmv — CHANGES_REQUESTED** [review]
# Impressive work, team! 🖥️ 👀 ✨ 
## However, some changes are still required... 🧰 🔨 🔧 

Your current tests are great. You cover both positive and negative cases. However, you seem to have missed out on a few details. Please read my code comments and fix these issues!

### Submit these changes and request a new code review! 😄 👍 

### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️

File: spec/game_logic_spec.rb (hunk 1 above)
Comment (alvarorf) [reply spec/game_logic_spec.rb:60]: You are right :bowtie:, we forgot about it and instead called the `rows` property from the `Board ` class.

**Review by maelfosso — APPROVED** [review]
#  Congratulations :clap: 

Thank you for applying the suggested changes :smiley: 
You can now merge it :+1: 

Good continuation :handshake: 

Mael FOSSO

**Pull request merged**